src/app/InteractionModelEngine.cpp

```diff
@@ -278,6 +278,37 @@ CHIP_ERROR InteractionModelEngine::ShutdownSubscription(const ScopedNodeId & aPe
     return CHIP_ERROR_KEY_NOT_FOUND;
 }
 
+bool InteractionModelEngine::IsSubjectSubscriptionActive(const FabricIndex & aFabricIndex, const NodeId & subjectID)
+{
+    for (uint16_t index = 0; index < GetNumActiveReadHandlers(); index++)
+    {
+        ReadHandler * readHandler = ActiveHandlerAt(index);
+        if (readHandler == nullptr)
+        {
+            // Should not happen
+            continue;
+        }
+
+        if (readHandler->IsType(ReadHandler::InteractionType::Subscribe))
```

Comment:
I would suggest reversing the test, returning/continuing early if not a subscribe handler, and outdenting the actual code.

Reply:
done